e2e/setup_teardown.go

```diff
@@ -9,22 +9,39 @@ func BeforeTestSuiteSetup() {
 	if lvmOperatorInstall {
 		debug("BeforeTestSuite: deploying LVM Operator\n")
 		err := DeployManagerObj.DeployLVMWithOLM(LvmCatalogSourceImage, LvmSubscriptionChannel)
-		debug("%v", err)
 		gomega.Expect(err).To(gomega.BeNil())
 	}
 
 	debug("BeforeTestSuite: starting LVM Cluster\n")
 	err := DeployManagerObj.StartLVMCluster()
-	debug("%v", err)
 	gomega.Expect(err).To(gomega.BeNil())
 
 	debug("BeforeTestSuite: creating Namespace %s\n", TestNamespace)
 	err = DeployManagerObj.CreateNamespace(TestNamespace)
-	debug("%v", err)
 	gomega.Expect(err).To(gomega.BeNil())
 
 	SuiteFailed = false
 
 	debug("------------------------------\n")
 
 }
+
+// AfterTestSuiteCleanup is the function called to tear down the test environment.
+func AfterTestSuiteCleanup() {
+
+	debug("\n------------------------------\n")
+
+	debug("AfterTestSuite: deleting Namespace %s\n", TestNamespace)
+	err := DeployManagerObj.DeleteNamespaceAndWait(TestNamespace)
```

Comment:
- a little bit confused here
1. does this just invoke delete ns and return immediately or it'll wait till ns is deleted
2. in both the cases, if ns deletion is invoked first then operator may get deleted leaving behind the lvmcluster which can't be removed unless the finalizer is removed
3. as lvmcluster cr will be left behind namespace removal will not happen at all

pls correct me if i'm misinterpreting the cleanup operation. I think below block `if lvmClusterUninstall ...` should be at the top of this function.

Reply:
This is test ns deletion, the ns in which we will be having pvc and other things. 
The ns that I used to install the resources in, is deleted in the end.

Does it sounds good?

Reply:
Yes, my mistake then, thanks for clarification.